openmmml/models/macepotential.py

```diff
@@ -0,0 +1,222 @@
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS, CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+from openmmml.mlpotential import MLPotential, MLPotentialImpl, MLPotentialImplFactory
+import openmm
+from typing import Iterable, Optional, Union, Tuple
+from openmmml.models.utils import simple_nl
+import logging
+
+
+class MACEPotentialImplFactory(MLPotentialImplFactory):
+    """This is the factory that creates MACEPotentialImpl objects."""
+
+    def createImpl(self, name: str, model_path: str, **args) -> MLPotentialImpl:
+        return MACEPotentialImpl(name, model_path)
+
+class MACEPotentialImpl(MLPotentialImpl):
+    """This is the MLPotentialImpl implementing the MACE potential.
+
+    The potential is implemented using MACE to build a PyTorch model.  A
+    TorchForce is used to add it to the OpenMM System.
+
+    You must specify the path to a trained MACE model e.g.
+    >>> potential = MLPotential('mace', model_path='MACE.model')
+
+    When you create the system can optionally change the dtype of the model
+    with the keyword argument dtype='float32' or dtype='float64'. The default behavoir is
+    to use the dtype of the loaded MACE model.
+    e.g.
+    >>> system = potential.createSystem(topology, dtype='float32')
+
+    You can also specify that the full atomic energy (including the atom self energy) is returned 
+    rather than the default of the interaction energy.
+    e.g.
+    >>> system = potential.createSystem(topology, interaction_energy=False)
+    
+    """
+
+    def __init__(self, name, model_path):
+        self.name = name
+        self.model_path = model_path
+
+    def addForces(self,
+                  topology: openmm.app.Topology,
+                  system: openmm.System,
+                  atoms: Optional[Iterable[int]],
+                  forceGroup: int,
+                  implementation : str=None,
+                  dtype: str=None,
+                  interaction_energy: bool=True,
+                  **args):
+        
+
+        import torch
+        import openmmtorch
+        from e3nn.util import jit
```

Comment:
This should give a better error message than import failed. An explanation of missing MACE and pip command should be shown.